app/controllers/distributions_controller.rb

```diff
@@ -164,4 +165,17 @@ def filter_params
 
     params.require(:filters).slice(:by_item_id, :by_partner)
   end
+
+  def find_distribution
+    Distribution.includes(:line_items).includes(:storage_location).find(params[:id])
+  end
+
+  def allow_edit?
+    distribution = find_distribution
+    unless distribution.issued_at > Time.zone.today || current_user.organization_admin.eql?(true)
+      flash[:error] = 'To edit a distribution,
+      you must be an organization admin or the current date must be major then today.'
+      redirect_to distributions_path
```

Comment:
A couple suggestions:

 1. predicates (methods ending in `?`) are generally expected to return either `truthy` or `falsey` values (eg. `true`, `false`, `nil`, etc -- things that can be placed into a boolean expression); they shouldn't produce side effects. *Could you change this one to just return true or false, depending on whether or not editing is allowed?*

 2. Setting the flash and doing the redirect should really be done in the action itself, instead of indirectly in the method, so that the flow of the action is clear. I think you can probably remove the `before_action` for this, and just evaluate `allow_edit?` in the action itself (since it's only used in one place, the indirection on `before_action` isn't really necessary. 

 3. Here's a cool Rails trick: when your model has a boolean attribute, you can treat it like a predicate (append a `?`) to the end -- so `current_user.organization_admin.eql?(true)` can become => `current_user.organization_admin?` :100: :slightly_smiling_face: